doc/boxes.el

```diff
@@ -35,76 +35,72 @@
     (emacs-lisp-mode . "lisp-cmt") (lisp-mode . "lisp-cmt"))
-  "The default comment type based on file names."
-:group 'customize
-)
+  "Default box type based on the major mode of the buffer."
+  :type '(alist :key-type symbol :value-type string)
+  :group 'boxes)
 
-;;;###autoload
-(defun boxes-set-default-type (mode)
-  "Set the default box mode according to the buffer's major mode."
-  (setq boxes-default-type (or (cdr (assoc mode boxes-known-modes)) "c-cmt2")))
+(defconst boxes-types-list
+  (let ((types (process-lines boxes-command "-q" "(all)")))
+    (mapcar (lambda(type) (replace-regexp-in-string " *\(alias\) *$" "" type)) types))
+  "List of types available to the current boxes implementation.")
 
-;;;###autoload
-(defvar boxes-default-type nil  "The default type of comment.")
-(make-variable-buffer-local 'boxes-default-type)
+(defvar boxes-history nil
+  "Boxes types history.")
+
+(defvar-local boxes-default-type nil
+  "The default type of box.")
+
+(defun boxes-default-type (mode)
+  "Get the default box type for the given buffer major MODE."
+  (or (cdr (assoc mode boxes-known-modes)) "c-cmt2"))
 
-;;;###autoload
-(defvar boxes-args ""
-  "Arguments to the boxes command.")
-(make-variable-buffer-local 'boxes-args)
 ;;;###autoload
 (defun boxes-create ()
   "Automagicly create a new box around the region based on the default type."
   (interactive "*")
   (boxes-command-on-region (region-beginning) (region-end) boxes-default-type))
 
-
 ;;;###autoload
 (defun boxes-remove ()
   "Automagicly remove a new box around the region based on the default type."
   (interactive "*")
   (boxes-command-on-region (region-beginning) (region-end) boxes-default-type 1))
 
-
 ;;;###autoload
 (defun boxes-command-on-region (start end type &optional remove)
-  "Create/Remove boxes from a region.  To create just select a region and M-x boxes-command-on-region
-then you will be prompted to enter a box type.  The type selection can use tab completion on the types available.
-To remove a box simply prefix a 1 to the callL M-1 M-x boxes-command-on-region will remove a box from a region."
+  "Create/Remove boxes from a region.
+To create just select a region and M-x boxes-command-on-region then you will be prompted to enter a box type.
+The type selection can use tab completion on the types available.
+To remove a box simply prefix a 1 to the call: M-1 M-x boxes-command-on-region will remove a box from a region."
   (interactive (let ((string
 		      (completing-read (format "Box type (%s): " boxes-default-type)
-				       boxes-types-alist nil t nil 'boxes-history boxes-default-type)))
+				       boxes-types-list nil t nil 'boxes-history boxes-default-type)))
 		(list (region-beginning) (region-end)
 		      string
 		      current-prefix-arg)))
-  (if type
-      (setq boxes-default-type type)
-    (setq boxes-default-type (boxes-set-default-type major-mode)
-          type boxes-default-type))
+  (when (or (null type) (string= type ""))
```

Comment:
‘Completing-read’ can return an empty string when there is no user input, so added a check for that.